### PR TITLE
Linkedin login and user data streamed to firebase firestore

### PR DESCRIPTION
**Pull request by justinlim1**
Once the user logs in to Linkedin and receives a response, the response is parsed and stored in Firebase Firestore database with the following fields: Linkedin ID, first name, last name, email.

### CODE REVIEW

File: ReferralPlease/ReferralPlease/ViewController.swift [hunk 1]
```diff
@@ -141,93 +158,136 @@ extension ViewController: WKNavigationDelegate {
...
+
+                    let accessToken = results?["access_token"] as? String
+                        print("accessToken is: \(accessToken ?? "")")
+
+                    let expiresIn = results?["expires_in"] as? Int
+                        print("expires in: \(expiresIn ?? nil)")
+
+                    // Get user's id, first name, last name, profile pic url
+                        if let access = accessToken{
+                    self.fetchLinkedInUserProfile(accessToken: access)
```
Comment (narrator0) [inline ReferralPlease/ReferralPlease/ViewController.swift]: I'm quite annoyed by the indentation of this block. Is it possible that you can fix that?

File: ReferralPlease/ReferralPlease/ViewController.swift [hunk 2]
```diff
@@ -141,93 +158,136 @@ extension ViewController: WKNavigationDelegate {
...
-                    */
-                    if let pictureUrls = linkedInProfileModel?.profilePicture.displayImage.elements[2].identifiers[0].identifier {
-                        linkedinProfilePic = pictureUrls
-                    } else {
-                        linkedinProfilePic = "Not exists"
+                    do {
+            
+    
+                        let linkedInProfileModel = try JSONDecoder().decode(LinkedInProfileModel.self, from: jsonData)
+                        self.db?.collection("users").document(linkedInProfileModel.id).setData([
```
Comment (narrator0) [inline ReferralPlease/ReferralPlease/ViewController.swift:239]: @justinlim1 if I login the second time, does this overwrite the old data or will it add a duplicate?

Comment (justinlim1) [reply ReferralPlease/ReferralPlease/ViewController.swift:239]: Its indexed by linkedin id so the document will just update unless linkedin id changes

Comment (narrator0) [reply ReferralPlease/ReferralPlease/ViewController.swift:239]: nice

**Review by narrator0 — APPROVED** [review]

**Review by subyun — APPROVED** [review]

**Pull request merged**